src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -1277,12 +1278,16 @@ public function getPricesAction(int $orderId): JsonResponse
     {
         /** @var OrderForViewing $orderForViewing */
         $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+        $orderForViewingPrices = $orderForViewing->getPrices();
+        $zeroNumber = new Number('0');
 
         return $this->json([
-            'orderTotalFormatted' => $orderForViewing->getPrices()->getTotalAmountFormatted(),
-            'productsTotalFormatted' => $orderForViewing->getPrices()->getProductsPriceFormatted(),
-            'shippingTotalFormatted' => $orderForViewing->getPrices()->getShippingPriceFormatted(),
-            'taxesTotalFormatted' => $orderForViewing->getPrices()->getTaxesAmountFormatted(),
+            'orderTotalFormatted' => $orderForViewingPrices->getTotalAmountFormatted(),
+            'discountsAmountFormatted' => $orderForViewingPrices->getDiscountsAmountFormatted(),
+            'discountsAmountDisplayed' => $orderForViewingPrices->getDiscountsAmountRaw()->isGreaterThan($zeroNumber),
```

Comment:
Maybe you should simply return `getDiscountsAmountRaw` It will provide more info than just a boolean (the test can be performed in js).
All those formatted prices are a bad idea in my opinion.. Very often we need to return the formatted price along with the raw one.. We have a CLDR component on front side as well but it's never been used sadly making the whole PHP code more heavy
But that's not the question here anyway 😅

Reply:
isn't it one of the the goals of "xxxForViewing" objects to give us ready to use values so that it doesn't have to be caculated in frontend?